lib/stitches/guided_fill.py

```diff
@@ -0,0 +1,355 @@
+    exist to build such a path, and we'll use Hierholzer's algorithm.
+
+    A graph must have an Eulerian Path if every node in the graph has an
+    even number of edges touching it.  Our goal here is to build a graph
+    that will have this property.
+
+    Based on the paper linked above, we'll build the graph as follows:
+
+        * nodes are the endpoints of the grating segments, where they meet
+        with the outer outline of the region the outlines of the interior
+        holes in the region.
+        * edges are:
+        * each section of the outer and inner outlines of the region,
+            between nodes
+        * double every other edge in the outer and inner hole outlines
+
+    Doubling up on some of the edges seems as if it will just mean we have
+    to stitch those spots twice.  This may be true, but it also ensures
+    that every node has 4 edges touching it, ensuring that a valid stitch
+    path must exist.
+    """
+
+    debug.add_layer("auto-fill fill stitch")
+
+    rows_of_segments = intersect_region_with_grating_guideline(shape, guideline, row_spacing)
+
+    # segments = [segment for row in rows_of_segments for segment in row]
+
+    graph = networkx.MultiGraph()
+
+    for i in range(len(rows_of_segments)):
+        for segment in rows_of_segments[i]:
+            # First, add the grating segments as edges.  We'll use the coordinates
+            # of the endpoints as nodes, which networkx will add automatically.
+
+            # networkx allows us to label nodes with arbitrary data.  We'll
+            # mark this one as a grating segment.
+            # graph.add_edge(*segment, key="segment", underpath_edges=[])
+            previous_neighbors = [(seg[0], seg[-1])
+                                  for seg in rows_of_segments[i-1] if i > 0]
+            next_neighbors = [(seg[0], seg[-1]) for seg in rows_of_segments[(i+1) %
+                              len(rows_of_segments)] if i < len(rows_of_segments)-1]
+
+            graph.add_edge(segment[0], segment[-1], key="segment", underpath_edges=[],
+                           geometry=LineString(segment), previous_neighbors=previous_neighbors, next_neighbors=next_neighbors,
+                           projected_points=DEPQ(iterable=None, maxlen=None), already_rastered=False)
+
+    tag_nodes_with_outline_and_projection(graph, shape, graph.nodes())
+    add_edges_between_outline_nodes(graph, duplicate_every_other=True)
+
+    if starting_point:
+        insert_node(graph, shape, starting_point)
+
+    if ending_point:
+        insert_node(graph, shape, ending_point)
+
+    debug.log_graph(graph, "graph")
+
+    return graph
+
+
+def get_segments(graph):
+    segments = []
+    for start, end, key, data in graph.edges(keys=True, data=True):
+        if key == 'segment':
+            segments.append(data["geometry"])
+
+    return segments
+
+
+def process_travel_edges(graph, fill_stitch_graph, shape, travel_edges):
```

Comment:
I see what you're trying to do by defining get_segments) and process_travel_edges() here, but it doesn't work.  These ones here are never run.  `guided_fill()` calls `build_travel_graph()` from auto_fill.py, and `build_travel_graph()` will call the `process_travel_edges()` that's in auto_fill.py.